### PR TITLE
Add ODE statistics postprocessor

### PR DESCRIPTION
**Pull request by jdannberg**
Note that this only works if the dealii version is from after https://github.com/dealii/dealii/pull/17096 was merged (Thank you for fixing this, @bangerth!). Otherwise this will output nan.

I have added the new postprocessor to a bunch of tests so that we'll see if iterations counts change in the future. This means I'll also still have to update all these tests.

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: source/material_model/grain_size.cc [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <aspect/heating_model/shear_heating.h>
 #include <aspect/material_model/rheology/visco_plastic.h>
 #include <aspect/utilities.h>
+#include <aspect/simulator.h>
```
Comment (tjhei) [inline source/material_model/grain_size.cc:28]: do you need the simulator include here? The signals header is not enough?

File: source/postprocess/ODE_statistics.cc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                                             total_iteration_count / number_of_solves
+                                             :
+                                             total_iteration_count;
+
+      statistics.add_value("Average iterations for ODE solver",
+                           average_iteration_count);
+#else
+      // The computation is not correct for dealii versions older than
+      // June 2024.
+      statistics.add_value("Average iterations for ODE solver",
```
Comment (tjhei) [inline source/postprocess/ODE_statistics.cc:86]: why don't you move the add_value() call below the #if/#else block by assigning ``const double average_iteration_count = std::numeric_limits<double>::quiet_NaN());``?

**Review by tjhei — APPROVED** [review]
very nice

File: source/material_model/grain_size.cc (hunk 1 above)
Comment (jdannberg) [reply source/material_model/grain_size.cc:28]: Yes, it doesn't compile without it.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**